demos/twa-webview-fallback/src/main/java/com/google/browser/examples/twawebviewfallback/WebViewFallbackActivity.java

```diff
@@ -139,6 +199,28 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
 
                 return super.shouldOverrideUrlLoading(view, request);
             }
+
+            private boolean matchExtraOrigins(Uri navigationUri) {
+                for (Uri uri: mExtraOrigins) {
+                    if (uriOriginsMatch(uri, navigationUri)) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            private boolean uriOriginsMatch(Uri uriA, Uri uriB) {
```

Comment:
I think you should also check that the ports match. And maybe instead of comparing the schemes, make sure they are both https?

Reply:
Added the check for `port`. I'm now checking for `https` in `launchUrl` when starting the Activity and ignoring additionalOrigins that are not `https`. Kept `getScheme().equals()` here.